packages/esbuild-plugin/src/index.ts

```diff
@@ -41,6 +43,36 @@ function esbuildReleaseInjectionPlugin(injectionCode: string): UnpluginOptions {
   };
 }
 
+function esbuildReactAnnotatePlugin(): UnpluginOptions {
+  const pluginName = "sentry-esbuild-react-annotate-plugin";
+  return {
+    name: pluginName,
+    esbuild: {
+      setup({ onLoad }) {
+        onLoad({ filter: /\.(t|j)sx$/ }, async (args) => {
+          const code = await fsPromises.readFile(args.path, "utf8");
+          const loader = args.path.endsWith(".tsx") ? "tsx" : "jsx";
+          const results = await createReactAnnotateHooks().transform(code, args.path);
+
+          let contents: string = code;
+
+          if (typeof results === "string") {
+            contents = results;
+          } else if (results?.code) {
+            contents = results.code;
+          }
+
+          return {
+            loader,
+            pluginName,
+            contents,
+          };
+        });
+      },
+    },
+  };
+}
```

Comment:
Unfortunately, we cannot use a simple `onLoad` hook for this funcitonality esbuild plugin, and in fact, I believe we cannot do this with good conscience in esbuild at all. Esbuild doesn't have a transform hook and that is by design. Read this comment I wrote for a similar situation: https://github.com/getsentry/sentry-javascript-bundler-plugins/pull/428#discussion_r1366770708

I don't feel like pressing the "changes requested" button but please consider this as a pretty hard blocker!

Reply:
Hey Luca, thanks for letting me know. I see, that's quite unfortunate that we can't use `onLoad` like this :/ 
Would you know of any alternatives that we can use to support this functionality in esbuild?

I assume the biggest problem here is that the `onLoad` hook only loads in files once, so this will prevent other loaders from running on these same files. In that case, can we fix this by using a `namespace`?

The esbuild docs says this:

> A callback added using onLoad will be run for each unique path/namespace pair that has not been marked as external.

It sounds like as long as we assign a unique namespace during the `onResolve` hook, we can get around this. Please lmk if this is an acceptable solution :)

Reply:
The plugin should not add much computational overhead as well, since it is an O(n) time algorithm. All it's doing is parsing JSX files and adding additional properties onto JSX elements

Reply:
I honestly do not think you can do this in esbuild with good conscience. If you assign a different namespace to the files in onResolve you will prevent the file from being picked up by other plugins' onLoad hooks and breaking them. This might be fine in a lot of cases but as soon as people add their own onLoad hooks this will break. I'll leave it up to you but just so you're aware.

Reply:
Understood, in that case we will hold off on releasing this plugin to esbuild for now, but we can reconsider it in the future if customers ask for it 👍